x-pack/solutions/chat/plugins/workchat-app/server/routes/agents.ts

```diff
@@ -131,7 +133,37 @@ export const registerAgentRoutes = ({ getServices, router, logger }: RouteDepend
 
       const agent = await client.update(agentId, payload);
 
-      return res.ok<CreateAgentResponse>({
+      return res.ok<UpdateAgentResponse>({
+        body: {
+          success: true,
+          agent,
+        },
+      });
+    })
+  );
+
+  router.delete(
+    {
+      path: '/internal/workchat/agents/{agentId}',
+      security: {
+        authz: {
+          requiredPrivileges: [apiCapabilities.manageWorkchat],
+        },
+      },
+      validate: {
+        params: schema.object({
+          agentId: schema.string(),
+        }),
+      },
+    },
+    wrapHandler(async (ctx, request, res) => {
+      const { agentId } = request.params;
+      const { agentService } = getServices();
+      const client = await agentService.getScopedClient({ request });
+
+      const agent = await client.delete(agentId);
+
+      return res.ok<DeleteAgentResponse>({
         body: {
           success: true,
           agent,
```

Comment:
we don't need to return deleted agent in the response

Reply:
done